contracts/BSCValidatorSet.sol

```diff
@@ -400,6 +433,52 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
     // averageDistribute*rest may less than income, but it is ok, the dust income will go to system reward eventually.
   }
 
+  /*********************** For Temporary Maintenance **************************/
+  function canEnterMaintenance(address validator, uint256 index) public view returns (bool can, uint256 exitMaintenanceReward) {
+    Validator memory validatorInfo = currentValidatorSet[index];
+
+    if (
+      (MaxNumOfMaintaining == 0 || MaxMaintainingTime == 0)         // - 1. check if not start
+      || maintainingValidatorSet.length >= MaxNumOfMaintaining      // - 2. check if exceeded upper limit
+      || validatorInfo.jailed                                       // - 3. check if jailed
+      || maintainInfoMap[validator].isMaintaining                   // - 4. check if maintaining
+      || maintainInfoMap[validator].maintainStartAt > 0             // - 5. check if has Maintained
+      || currentValidatorSet.length <= 1                            // - 6. check num of remaining currentValidators
```

Comment:
damn. I find a bug in previous implement, `currentValidatorSet.length <= 1` need exclude the validator that is in jail and in maintaining .

Reply:
You need change the implementation  of `getValidators` and use `len(getValidators()) <=1` as condition here.

Reply:
fixed